### PR TITLE
add a /roll command

### PR DESCRIPTION
**Pull request by yoganc**
add a /roll command that takes either a numeric max value for [1,max] or <numDice>d<maxValue> for [1, maxValue] numDice times.

### CODE REVIEW

File: OhHell/src/client/GameCanvas.java [hunk 1]
```diff
@@ -1505,6 +1505,30 @@ public void keyReleased(KeyEvent e) {
...
                                 }
+                            } else if (command.equals("roll") && commandContent.length == 2) {
+                                int maxValue;
+                                int numDice = 1;
+                                try {
+                                    maxValue = Integer.parseInt(content);
+                                } catch (NumberFormatException e) {
+                                    List<String> split = content.split("d", 2);
+                                    try {
+                                        maxValue = Integer.parseInt(split.get(0));
```
Comment (campbellsoup37) [inline OhHell/src/client/GameCanvas.java]: I'm guessing this one should be split.get(1)

File: OhHell/src/client/GameCanvas.java [hunk 2]
```diff
@@ -1505,6 +1505,30 @@ public void keyReleased(KeyEvent e) {
...
+                                        numDice = Integer.parseInt(split.get(0));
+                                    } catch (NumberFormatException | IndexOutOfBoundsException e2) {
+                                        numDice = 0;
+                                    }
+                                }
+                                if (numDice > 0) {
+                                    int rndValue = numDice;
+                                    for (int i = 0; i < numDice; ++i) {
+                                        rndValue += random.nextInt(maxValue);
```
Comment (campbellsoup37) [inline OhHell/src/client/GameCanvas.java:1525]: This will roll between 0 and maxValue - 1. Maybe random.nextInt(maxValue) + 1?

**Review by campbellsoup37 — APPROVED** [review]

File: OhHell/src/client/GameCanvas.java (hunk 2 above)
Comment (campbellsoup37) [reply OhHell/src/client/GameCanvas.java:1525]: never mind I see that rndValue starts at numDice.

File: OhHell/src/client/GameCanvas.java (hunk 1 above)
Comment (campbellsoup37) [reply OhHell/src/client/GameCanvas.java]: A couple other things -- the variable name e is already used as the KeyEvent. And content.split will return a String[], not a List<String>, right?

Comment (yoganc) [reply OhHell/src/client/GameCanvas.java]: That's what I get for not using an IDE where I can check whether it builds. Fixed all 3 issues, PTAL.

Comment (campbellsoup37) [reply OhHell/src/client/GameCanvas.java]: I think the numbers are switched now

Comment (yoganc) [reply OhHell/src/client/GameCanvas.java]: corrected

**Pull request merged**